packages/fsengagement/src/EngagementComp.tsx

```diff
@@ -418,5 +419,9 @@ export default function(
         </View>
       );
     }
+
+    dataKeyExtractor = (item: BlockItem): string => {
+      return item.id || item.key || Math.floor(Math.random() * 1000000).toString();
```

Comment:
this part is where I was a little unsure of the data structures returned from the API so I tried my best to use the values I felt uniquely identified each part, falling back to the previous random key which might not be the best idea but I left it in

Reply:
Yeah im pretty sure it's always `key` now, looking at the JSON type i see id is in there too though.  This should be fine.